### PR TITLE
Kubemark scripts retry failed gcloud commands

### LINKED ISSUE #18334: Kubemark - add retries in scripts.

### PR DESCRIPTION
**Pull request by gmarek**
Fixes #18334
Redone #20158.

cc @mwielgus @fgrzadkowski @ihmccreery

### CODE REVIEW

File: test/kubemark/start-kubemark.sh [hunk 1]
```diff
@@ -46,39 +46,39 @@ cd $CURR_DIR
...
-    --scopes "storage-ro,compute-rw,logging-write" \
-    --disk "name=${MASTER_NAME}-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no"
-
-gcloud compute firewall-rules create "${INSTANCE_PREFIX}-kubemark-master-https" \
-    --project "${PROJECT}" \
-    --network "${NETWORK}" \
-    --source-ranges "0.0.0.0/0" \
-    --target-tags "${MASTER_TAG}" \
-    --allow "tcp:443"
+run-gcloud-compute disks create "${MASTER_NAME}"-pd \
```
Comment (ihmccreery) [inline test/kubemark/start-kubemark.sh]: I'd expect this to be `"${MASTER_NAME}-pd"`; why switch it?

File: test/kubemark/common.sh [hunk 2]
```diff
@@ -23,3 +24,22 @@ export PROJECT
...
 MASTER_TAG="kubemark-master"
 EVENT_STORE_NAME="${INSTANCE_PREFIX}-event-store"
+
+RETRIES=3
+
+# Runs gcloud compute command with the given parameters. Up to $RETRIES will be made
+# to execute the command.
+# arguments:
+# $@: all stuff that goes after 'gcloud compute '
+function run-gcloud-compute {
```
Comment (ihmccreery) [inline test/kubemark/common.sh]: `run-gcloud-compute-with-retries`?

File: test/kubemark/start-kubemark.sh [hunk 3]
```diff
@@ -46,39 +46,39 @@ cd $CURR_DIR
...
+  ${GCLOUD_COMMON_ARGS} \
+  --machine-type "${MASTER_SIZE}" \
+  --image-project="${MASTER_IMAGE_PROJECT}" \
+  --image "${MASTER_IMAGE}" \
+  --tags "${MASTER_TAG}" \
+  --network "${NETWORK}" \
+  --scopes "storage-ro,compute-rw,logging-write" \
+  --disk "name=${MASTER_NAME}-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no"
+
+run-gcloud-compute firewall-rules create ${INSTANCE_PREFIX}-kubemark-master-https \
```
Comment (ihmccreery) [inline test/kubemark/start-kubemark.sh]: Why remove quotes?

**Comment by ihmccreery** [comment]
LGTM once nits are addressed.

File: test/kubemark/start-kubemark.sh (hunk 1 above)
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: Done.

File: test/kubemark/start-kubemark.sh (hunk 3 above)
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: Done.

File: test/kubemark/common.sh (hunk 2 above)
Comment (gmarek) [inline test/kubemark/common.sh]: Done.

**Comment by gmarek** [comment]
@ihmccreery PTAL

**Comment by ihmccreery** [comment]
LGTM, thanks @gmarek.

**Pull request merged**